### PR TITLE
All services can send files by email if they have set contact_link

### PR DESCRIPTION
**Pull request by CrystalPea**
This PR changes validation away from `upload_document` service permission and towards checking if service has provided contact details for the feature.

Connected to PR: https://github.com/alphagov/notifications-admin/pull/3307 (they should go out around the same time)

Story ticket: https://www.pivotaltracker.com/story/show/171254127

### CODE REVIEW

File: app/notifications/validators.py [hunk 1]
```diff
@@ -85,7 +85,13 @@ def service_has_permission(notify_type, permissions):
...
     if not service_has_permission(notify_type, permissions):
         raise BadRequestError(message="Service is not allowed to send {}".format(
-            get_public_notify_type_text(notify_type, plural=True)))
+            get_public_notify_type_text(notify_type, plural=True)
+        ))
+
+
+def check_if_service_can_send_files_by_email(service_contact_link):
+    if not service_contact_link:
+        raise BadRequestError(message="Go to Service Settings to turn on sending files by email")
```
Comment (servingUpAces) [inline app/notifications/validators.py]: Should this message talk about setting the service contract details?

Comment (CrystalPea) [reply app/notifications/validators.py]: We discussed it with Karl and settled at: `"Send files by email is not set up yet. Go to your settings page to manage send files by email"`

Then users know where to go (the tab in settings is called `Send files by email`) and they get all the guidance they need once they get there

File: app/notifications/validators.py [hunk 2]
```diff
@@ -85,7 +85,15 @@ def service_has_permission(notify_type, permissions):
...
         raise BadRequestError(message="Service is not allowed to send {}".format(
-            get_public_notify_type_text(notify_type, plural=True)))
+            get_public_notify_type_text(notify_type, plural=True)
+        ))
+
+
+def check_if_service_can_send_files_by_email(service_contact_link):
+    if not service_contact_link:
+        raise BadRequestError(
+            message="Send files by email is not set up yet. Go to your settings page to manage send files by email"
```
Comment (karlchillmaid) [inline app/notifications/validators.py]:
```suggestion
            message="Send files by email has not been set up. Go to your Settings page to manage Send files by email."
```

File: tests/app/notifications/test_validators.py [hunk 3]
```diff
@@ -528,3 +529,17 @@ def test_check_reply_to_sms_type(sample_service):
 def test_check_reply_to_letter_type(sample_service):
     letter_contact = create_letter_contact(service=sample_service, contact_block='123456')
     assert check_reply_to(sample_service.id, letter_contact.id, LETTER_TYPE) == '123456'
+
+
+def test_check_if_service_can_send_files_by_email_raises_if_no_contact_link_set(sample_service):
+    with pytest.raises(BadRequestError) as e:
+        check_if_service_can_send_files_by_email(sample_service.contact_link)
+
+    expected_message = "Send files by email is not set up yet. Go to your settings page to manage send files by email"
```
Comment (karlchillmaid) [inline tests/app/notifications/test_validators.py]:
```suggestion
    expected_message = "Send files by email has not been set up. Go to your Settings page to manage Send files by email."
```

File: app/notifications/validators.py [hunk 4]
```diff
@@ -85,7 +85,15 @@ def service_has_permission(notify_type, permissions):
...
         raise BadRequestError(message="Service is not allowed to send {}".format(
-            get_public_notify_type_text(notify_type, plural=True)))
+            get_public_notify_type_text(notify_type, plural=True)
+        ))
+
+
+def check_if_service_can_send_files_by_email(service_contact_link):
+    if not service_contact_link:
+        raise BadRequestError(
+            message="Send files by email has not been set up. Go to your Settings page to manage Send files by email."
```
Comment (karlchillmaid) [inline app/notifications/validators.py]:
```suggestion
            message="Send files by email has not been set up - go to your Settings page to manage Send files by email"
```

**Pull request merged**